pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1616,6 +1647,50 @@ func configAdapters(ctx *domainContext, config types.DomainConfig) error {
 	return nil
 }
 
+func writeMountPointsToFile(mountPoints []MountPoints, status types.DomainStatus, file *os.File) error {
+	if len(mountPoints) != len(status.DiskStatusList){
+		err := fmt.Errorf("writeMountPointsToFile: Number of source: %v and target: %v mismatch.",
+			len(status.DiskStatusList), len(mountPoints))
+		log.Errorf(err.Error() )
+		return err
+	}
+
+	for i, mp := range mountPoints {
+		if mp.fileSystem == "" {
+			mp.fileSystem = "vfat"
+		} else {
+			if _, ok := FileSystems[mp.fileSystem]; !ok {
+				err := fmt.Errorf("writeMountPointsToFile: Invalid/unrecognized filesystem: %v", mp.fileSystem)
+				log.Errorf(err.Error())
+				return err
+			}
+		}
+		if mp.partition == 0 { //Assuming if mp.partition = 0, then it is not set
+			mp.partition = 1
+		} else {
+			if mp.partition > 4 || mp.partition < 0 {
+				err := fmt.Errorf("writeMountPointsToFile: Invalid partition: %v. Must be between 1 - 4", mp.partition)
```

Comment:
What are these numbers?? What does 0 -3 signify?

Reply:
fdisk(which we are using to create a partition on the image uploaded) creates partition 1-4. Expected a user input to specify to which partition should we mount the target. Removed this since this is out of scope.